crates/ruff_linter/src/rules/pyupgrade/rules/unnecessary_coding_comment.rs

```diff
@@ -57,46 +67,85 @@ pub(crate) fn unnecessary_coding_comment(
 ) {
     // The coding comment must be on one of the first two lines. Since each comment spans at least
     // one line, we only need to check the first two comments at most.
-    for comment_range in comment_ranges.iter().take(2) {
-        // If leading content is not whitespace then it's not a valid coding comment e.g.
-        // ```
-        // print(x) # coding=utf8
-        // ```
-        let line_range = locator.full_line_range(comment_range.start());
-        if !locator
-            .slice(TextRange::new(line_range.start(), comment_range.start()))
-            .trim()
-            .is_empty()
-        {
-            continue;
-        }
+    let coding_comments = comment_ranges
+        .iter()
+        .take(2)
+        .map(|comment_range| coding_comment(locator, indexer, *comment_range))
+        .collect::<Vec<_>>();
```

Comment:
It would be great if we can avoid collecting the comments here to avoid the allocation. I think doing so should also reduce the amount of code needed below.

```rust
let mut comments = comment_ranges.iter().take(...)...

let Some(first) = comments.next() else {
	return None;
}

if let Some(CodingComment::Other) = comments.next() {
	return None;
}

```